### ISSUE TITLE
Surpress 404/routing through PHP when looking for /favicon.ico

### ISSUE THREAD

**Issue by mariuswilms** [atelierdisko/hoi#17]
Some clients do not respect a favicon path given in the HTML document head section and continue to look for favicon.ico. This creates a lot of noise in the logs when that file is missing. It creates even more noise when any file that is non-existent is routed through a PHP front controller.

This should preferably be solved by a new NGINX rule in includes/app.conf (see below). It however will prevent from actually using a favicon.ico, as the rule blindly re-routes all traffic.

```
# Get rid of noisy searches by clients not respecting HTML head favicons.
location /favicon.ico {
  return 204;
}
```

**Comment by d1rk** [comment]
Why not just omit logs for these kind of requests? I have something similar to this, which would still work fine if a favicon.ico would be present (amongst other things):

```
##########
# do not serve and/or omit logging requests to expectable endpoints

location = /robots.txt  { access_log off; log_not_found off; }
location = /favicon.ico { access_log off; log_not_found off; }
location ~ /\.          { access_log off; log_not_found off; deny all; }
location ~ ~$           { access_log off; log_not_found off; deny all; }
location ~* (\.(bak|config|sql|fla|psd|ini|log|sh|inc|swp|dist)|~)$ {
  access_log off; log_not_found off; deny all;
}
```

**Comment by mariuswilms** [comment]
Thanks for submitting help.

I'm now preventing routing through the backend for favicon.ico. I've opted for not doing the same for robots.txt as that can also be generated dynamically through the app (we do that). 

I didn't implement support for the other rules you've mentioned, my reasoning is as follows: these are security related and use blacklisting, supporting them by default would wrongly suggest we take care of _all_ these kinds of threads, which would generate confidence, where we can't give some. Blacklists are often never complete, there are just too many different cases of confidential information.

I've also opted for the more common 404 and now allow the application to actually have a favicon.ico if it wishes to do so.
